### PR TITLE
Fix graphics flickering/stretching on Windows

### LINKED ISSUE #366: [Windows] Bad graphical flickering in Windowed mode
cboeGif

Windowed mode is screwy on Windows. I tried it with and without UI scaling. This also affects the scenario and character editors.

The gif I recorded doesn't capture the speed at which the flickering really happens, which is actually much faster.

### PR DESCRIPTION
**Pull request by NQNStudios**
Fixes #366 

This bug was unbelievably hard to track down and fix.

I think it was probably caused by a change in SFML's code for handling window resize events on Windows. When we add the menubar to the window by bypassing SFML and using the Windows APIs, SFML now detects that change and shrinks the logical viewport of your window without expanding the window itself to fit the menubar in it.

I updated `menuProc()` to catch that first unwanted resize event and discard it. I also made the window expand to fit the menubar's additional height, handled for a rare edge case where the menubar needs more than 1 row for its buttons, and made sure the splash screens still render in the right places.

I need to test this on Linux and Mac to make sure it didn't break those.

### CODE REVIEW

**Comment by NQNStudios** [comment]
This doesn't break the Linux behavior.

File: src/game/boe.startup.cpp [hunk 1]
```diff
@@ -109,9 +109,10 @@ bool handle_startup_press(location the_point) {
 
 void handle_splash_events() {
 	sf::Event event;
-	if(!mainPtr.pollEvent(event)) return;
-	if(event.type == sf::Event::GainedFocus || event.type == sf::Event::MouseMoved)
-		set_cursor(sword_curs);
+	while (mainPtr.pollEvent(event)) {
```
Comment (CelticMinstrel) [inline src/game/boe.startup.cpp:112]: No space after while please.

File: src/tools/winutil.win.cpp [hunk 2]
```diff
@@ -444,5 +444,11 @@ ModalSession::~ModalSession() {
 }
 
 int getMenubarHeight() {
-	return GetSystemMetrics(SM_CYMENU);
+	MENUBARINFO info;
+	info.cbSize = sizeof(MENUBARINFO);
+	if (GetMenuBarInfo(mainPtr.getSystemHandle(), OBJID_MENU, 0, &info))
```
Comment (CelticMinstrel) [inline src/tools/winutil.win.cpp:449]: Ditto.

**Comment by NQNStudios** [comment]
I checked and this also doesn't break anything on Mac OS.

**Comment by NQNStudios** [comment]
These changes didn't fix the scenario and character editors, though...

**Comment by NQNStudios** [comment]
Would be bad practice to leave something broken in the editors when there's a known solution, so I applied the same fix to them and it worked.

**Comment by NQNStudios** [comment]
I think issue #59 must be related to the same underlying cause (stuffing a menu bar into an SFML window making it squish its effective viewport weirdly). So I might as well see if that's still a problem and  try to fix that too.

**Comment by NQNStudios** [comment]
@CelticMinstrel I'm done fiddling with this one if you don't have any other changes in mind.

**Comment by CelticMinstrel** [comment]
I assume you checked that this doesn't break anything on Mac or Linux, right?

**Comment by NQNStudios** [comment]
Yeah I did
